irohad/ametsuchi/impl/storage_impl.hpp

```diff
@@ -159,9 +165,14 @@ namespace iroha {
       bool storeBlock(
           std::shared_ptr<const shared_model::interface::Block> block);
 
+      /**
+       * Method tries to perform rollback on passed session
+       */
+      void tryRollback(soci::session &session);
+
       std::unique_ptr<KeyValueStorage> block_store_;
 
-      std::shared_ptr<soci::connection_pool> connection_;
```

Comment:
It might be useful to have `std::shared_ptr<soci::connection_pool> &connection_` which will be a reference to connection pool field in wrapper, in order to simplify access in cpp file.